File: Cpp/2789_largest_element_in_an_array_after_merge_operations.cpp
// https://leetcode.cn/problems/largest-element-in-an-array-after-merge-operations/
        
class Solution {
public:
    long long maxArrayValue(vector<int>& nums) {
        long long res = nums.back();
        for (int i = nums.size()-2; i >= 0; --i) {
            if (nums[i] <= res) res += nums[i];
            else res = nums[i];
        }
        return res;
    }
};

File: Python3/2789_largest_element_in_an_array_after_merge_operations.py
# https://leetcode.cn/problems/largest-element-in-an-array-after-merge-operations/

class Solution:
    def maxArrayValue(self, nums):
        """
        Inputs:
            nums: list[int]
        Outputs:
            res: int
        """
        res = nums[-1]
        for i in range(len(nums)-2, -1, -1):
            if nums[i] <= res:
                res += nums[i]
            else:
                res = nums[i]
        return res